var/spack/repos/builtin/packages/exasp2/package.py

```diff
@@ -0,0 +1,94 @@
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Exasp2(MakefilePackage):
+    """ExaSP2 is a reference implementation of typical linear algebra algorithms
+    and workloads for a quantum molecular dynamics (QMD) electronic structure
+    code. The algorithm is based on a recursive second-order Fermi-Operator
+    expansion method (SP2) and is tailored for density functional based
+    tight-binding calculations of material systems. The SP2 algorithm variants
+    are part of the Los Alamos Transferable Tight-binding for Energetics
+    (LATTE) code, based on a matrix expansion of the Fermi operator in a
+    recursive series of generalized matrix-matrix multiplications. It is
+    created and maintained by Co-Design Center for Particle Applications
+    (CoPA). The code is intended to serve as a vehicle for co-design by
+    allowing others to extend and/or reimplement as needed to test performance
+    of new architectures, programming models, etc."""
+
+    tags = ['proxy-app']
+
+    homepage = "https://github.com/ECP-copa/ExaSP2"
+
+    version('develop', git='https://github.com/ECP-copa/ExaSP2',
+            branch='master')
+
+    variant('mpi', default=True, description='Build With MPI Support')
+
+    depends_on('bml')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('mpi', when='+mpi')
+    depends_on('bml@1.2.3:+mpi', when='+mpi')
+
+    build_directory = 'src'
+
+    @property
+    def build_targets(self):
+        targets = []
+        spec = self.spec
+        if '+mpi' in spec:
+            targets.append('PARALLEL=MPI')
+            targets.append('MPICC={0}'.format(spec['mpi'].mpicc))
+            targets.append('MPI_LIB=-L{0} -lmpi'.format(spec['mpi'].prefix.lib))
+            targets.append('MPI_INCLUDE=-I{0}'.format(spec['mpi'].prefix.include))
+        else:
+            targets.append('PARALLEL=NONE')
+        # NOTE: no blas except for mkl has been properly tested. OpenBlas was
+        #   briefly but not rigoruously tested. Using generic blas approach to
+        #   meet Spack requirements
+        targets.append('BLAS=GENERIC_SPACKBLAS')
+        math_libs = str(spec['lapack'].libs)
+        math_libs += ' ' + str(spec['lapack'].libs)
+        targets.append('SPACKBLASLIBFLAGS=' + math_libs)
+        math_includes = spec['lapack'].prefix.include
+        math_includes += " -I" + spec['blas'].prefix.include
```

Comment:
up to you if you try this, but there is not very-well documented feature of `HeaderList`. 
```
math_header = spec['lapack'].headers + spec['blas'].headers
math_header.cpp_flags
```
should give what you want. See http://spack.readthedocs.io/en/latest/llnl.util.html?highlight=HeaderList#llnl.util.filesystem.HeaderList

It is used in some packages already:
```
$ ag "cpp_flags" var/spack/repos/builtin/packages/
var/spack/repos/builtin/packages/abinit/package.py
152:            'with_libxc_incs={0}'.format(libxc.headers.cpp_flags),
164:                '--with-netcdf-incs={0}'.format(netcdff.headers.cpp_flags),

var/spack/repos/builtin/packages/cp2k/package.py
98:                spec['fftw'].headers.cpp_flags
115:                spec['fftw'].headers.cpp_flags
193:                    # spec['elpa:fortran'].headers.cpp_flags
200:                    # spec[pexsi:fortran].headers.cpp_flags

var/spack/repos/builtin/packages/environment-modules/package.py
62:        cpp_flags = ['-DUSE_INTERP_ERRORLINE']
77:            'CPPFLAGS=' + ' '.join(cpp_flags)

var/spack/repos/builtin/packages/hdf5/package.py
267:            cc(*(['-c', "check.c"] + spec['hdf5'].headers.cpp_flags.split()))

var/spack/repos/builtin/packages/openblas/package.py
194:        include_flags = spec['openblas'].headers.cpp_flags

var/spack/repos/builtin/packages/pocl/package.py
107:            cflags = spec["pocl"].headers.cpp_flags.split()

var/spack/repos/builtin/packages/superlu-dist/package.py
77:                spec['parmetis'].headers.cpp_flags,
78:                spec['metis'].headers.cpp_flags,
```